app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -33,6 +33,12 @@ cat << EOF > $SSL_CA_CERT
 <%= foreman_server_ca_cert %>
 EOF
 
+if [ -f /usr/bin/hostname ]; then
+  hostname=$(hostname -f)
+else
+  hostname=$(cat /etc/HOSTNAME);
```

Comment:
This doesn't exist
```suggestion
  hostname=$(cat /etc/hostname)
```

Also, there's no guarantee that the file does exist. For example, on EL6 and older it doesn't. On non-Linux it doesn't. I would add a check that the file exists.